File: 5_Longest_Palindromic_Substring/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def isPalindrome(sub: str) -> bool:
            return sub == sub[::-1]

        n = len(s)
        max_palindrome = s[0]

        for i in range(n):
            for j in range(i, n):
                substring = s[i : j + 1]
                if isPalindrome(substring):
                    if len(substring) > len(max_palindrome):
                        max_palindrome = substring

        return max_palindrome

File: 5_Longest_Palindromic_Substring/solution.c
bool isPalindrome(char* sub, int len) {
    for (int i = 0; i < len / 2; i++) if (sub[i] != sub[len - 1 - i]) return false;
    return true;
}
char* longestPalindrome(char* s) {
    int n = strlen(s), max_len = 1, start = 0;

    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            int len = j - i + 1;
            if (isPalindrome(s + i, len)) {
                if (len > max_len) { max_len = len; start = i; }
            }
        }
    }

    char* result = (char*)malloc((max_len + 1) * sizeof(char));
    strncpy(result, s + start, max_len);
    result[max_len] = '\0';
    return result;
}